File: project/06-Async/solution-code/app/src/main/java/ch/epfl/sweng/project/MainActivity.java
package ch.epfl.sweng.project;

import android.content.Intent;
import android.os.Bundle;
import android.view.inputmethod.EditorInfo;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Get a reference to the UI components
        EditText nameText = findViewById(R.id.mainName);
        Button goButton = findViewById(R.id.mainGoButton);
        Button weatherButton = findViewById(R.id.weatherButton);

        // Set the behaviour of the button
        goButton.setOnClickListener(clicked -> {
            String name = nameText.getText().toString();
            sayHello(name);
        });

        weatherButton.setOnClickListener(clicked -> {
            startActivity(new Intent(this, WeatherActivity.class));
        });

        // Bonus: trigger the button when the user presses "enter" in the text field
        nameText.setOnEditorActionListener((v, actionId, event) -> {
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                goButton.callOnClick();
                return true;
            }
            return false;
        });
    }

    /**
     * Opens a new {@link GreetingActivity} saying hello to the specified {@code userName}
     *
     * @param userName the user to greet
     */
    protected void sayHello(String userName) {
        Intent intent = new Intent(this, GreetingActivity.class);
        intent.putExtra(GreetingActivity.EXTRA_USER_NAME, userName);
        startActivity(intent);
    }
}

File: exercises/devops/sample-app-without-ci/app/src/main/java/ch/epfl/sweng/project/MainActivity.kt
package ch.epfl.sweng.project

import android.content.Intent
import android.os.Bundle
import android.view.KeyEvent
import android.view.inputmethod.EditorInfo
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

open class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Get a reference to the UI components
        val nameText = findViewById<EditText>(R.id.mainName)
        val goButton = findViewById<Button>(R.id.mainGoButton)
        val weatherButton = findViewById<Button>(R.id.weatherButton)

        // Set the behaviour of the button
        goButton.setOnClickListener {
            val name = nameText.text.toString()
            sayHello(name)
        }
        weatherButton.setOnClickListener {
            startActivity(
                Intent(
                    this,
                    WeatherActivity::class.java
                )
            )
        }

        // Bonus: trigger the button when the user presses "enter" in the text field
        nameText.setOnEditorActionListener { _: TextView?, actionId: Int, _: KeyEvent? ->
            if (actionId == EditorInfo.IME_ACTION_DONE) {
                goButton.callOnClick()
                return@setOnEditorActionListener true
            }
            false
        }
    }

    /**
     * Opens a new [GreetingActivity] saying hello to the specified `userName`
     *
     * @param userName the user to greet
     */
    private fun sayHello(userName: String?) {
        val intent = Intent(this, GreetingActivity::class.java)
        intent.putExtra(GreetingActivity.EXTRA_USER_NAME, userName)
        startActivity(intent)
    }
}